src/main/java/net/sf/mzmine/modules/visualization/spectra/simplespectra/mirrorspectra/MirrorScanWindow.java

```diff
@@ -103,24 +112,96 @@ public void setScans(PeakListRow row, SpectralDBPeakIdentity db) {
     DataPoint[] dpsA = scan.getDataPoints();
 
     //
-    double precursorMZB = db.getEntry().getPrecursorMZ();
+    Double precursorMZB = db.getEntry().getPrecursorMZ();
     Double rtB = (Double) db.getEntry().getField(DBEntryField.RT).orElse(0d);
     DataPoint[] dpsB = db.getEntry().getDataPoints();
-    this.setScans("", precursorMZA, rtA, dpsA, "database", precursorMZB, rtB, dpsB);
+
+
+    contentPane.removeAll();
+    // create without data
+    mirrorSpecrumPlot = SpectrumChartFactory.createMirrorChartPanel(
+        "Query: " + scan.getScanDefinition(), precursorMZA, rtA, null, "Library: " + db.getName(),
+        precursorMZB == null ? 0 : precursorMZB, rtB, null, false, true);
+    mirrorSpecrumPlot.setMaximumDrawWidth(4200);
+    mirrorSpecrumPlot.setMaximumDrawHeight(2500);
+    // add data
+    DataPointsTag[] tags =
+        new DataPointsTag[] {DataPointsTag.ORIGINAL, DataPointsTag.FILTERED, DataPointsTag.ALIGNED};
+    Color[] colors = new Color[] {Color.black, new Color(0xF57C00), new Color(0x388E3C)};

```

Comment:
This should definitely be defined as a public static final constant, with comments on what the hex colors are.